docs/development.md

```diff
@@ -35,3 +35,39 @@ Since this is very unlikely and most times an abs path is not strictly necessary
 "Should" functions can be used anywhere,
 but care should be taken not to overuse them when a fallback value is not sufficient.
 "Must" variants should be favored in testing code, since they will more obviously fail tests.
+
+## Cross-Platform Scripting
+
+We are writing bash scripts with the intention of running them in a POSIX environment.
+However, differences between operating systems, particularly between Linux and BSD,
+can still affect a script's behavior. 
+This section highlights key considerations to ensure your scripts run consistently across platforms.
+
+### Regular Expressions
+
+Differences between how Linux and BSD handle different standards of regular expressions can cause inconsistent outcomes when writing scripts.
+Here are three common regex standards you may encounter, along with key considerations for each:
+
+- **BRE (Basic Regular Expressions)**
+  - **Usage**: Default in many tools.
+  - **Consideration**: While BREs cover most common use cases, they lack more advanced features could simplify and lower the chance of compatibility issues with complex regular expressions.
+  - **[More Info](https://en.wikibooks.org/wiki/Regular_Expressions/Basic_Regular_Expressions)**
```

Comment:
I would prefer wikipedia links of wikibooks links.
The wikibooks one is more concise, but wikipedia often has more relevant information,
e.g. https://en.wikipedia.org/wiki/Regular_expression#IEEE_POSIX_Standard .